cmd/agent/app/reporter/grpc/flags.go

```diff
@@ -46,13 +46,17 @@ func AddFlags(flags *flag.FlagSet) {
 }
 
 // InitFromViper initializes Options with properties retrieved from Viper.
-func (b *ConnBuilder) InitFromViper(v *viper.Viper) *ConnBuilder {
+func (b *ConnBuilder) InitFromViper(v *viper.Viper) (*ConnBuilder, error) {
 	hostPorts := v.GetString(collectorHostPort)
 	if hostPorts != "" {
 		b.CollectorHostPorts = strings.Split(hostPorts, ",")
 	}
 	b.MaxRetry = uint(v.GetInt(retry))
-	b.TLS = tlsFlagsConfig.InitFromViper(v)
 	b.DiscoveryMinPeers = v.GetInt(discoveryMinPeers)
-	return b
+	if TLS, err := tlsFlagsConfig.InitFromViper(v); err != nil {
+		return b, err
```

Comment:
* not sure if this will pass the linter, if not you can swap the condition and do a return in the `else` part.
* it looks odd when local var is upper case, use `tls` or `cfg`